test/extensions/filters/http/common/jwks_fetcher_test.cc

```diff
@@ -0,0 +1,209 @@
+#include <chrono>
+#include <thread>
+
+#include "common/http/message_impl.h"
+#include "common/protobuf/utility.h"
+
+#include "extensions/filters/http/common/jwks_fetcher.h"
+
+#include "test/mocks/http/mocks.h"
+#include "test/mocks/server/mocks.h"
+#include "test/test_common/utility.h"
+
+using ::envoy::api::v2::core::HttpUri;
+using ::google::jwt_verify::Status;
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Common {
+namespace {
+
+const char publicKey[] = R"(
+{
+  "keys": [
+    {
+      "kty": "RSA",
+      "alg": "RS256",
+      "use": "sig",
+      "kid": "62a93512c9ee4c7f8067b5a216dade2763d32a47",
+      "n": "up97uqrF9MWOPaPkwSaBeuAPLOr9FKcaWGdVEGzQ4f3Zq5WKVZowx9TCBxmImNJ1qmUi13pB8otwM_l5lfY1AFBMxVbQCUXntLovhDaiSvYp4wGDjFzQiYA-pUq8h6MUZBnhleYrkU7XlCBwNVyN8qNMkpLA7KFZYz-486GnV2NIJJx_4BGa3HdKwQGxi2tjuQsQvao5W4xmSVaaEWopBwMy2QmlhSFQuPUpTaywTqUcUq_6SfAHhZ4IDa_FxEd2c2z8gFGtfst9cY3lRYf-c_ZdboY3mqN9Su3-j3z5r2SHWlhB_LNAjyWlBGsvbGPlTqDziYQwZN4aGsqVKQb9Vw",
+      "e": "AQAB"
+    },
+    {
+      "kty": "RSA",
+      "alg": "RS256",
+      "use": "sig",
+      "kid": "b3319a147514df7ee5e4bcdee51350cc890cc89e",
+      "n": "up97uqrF9MWOPaPkwSaBeuAPLOr9FKcaWGdVEGzQ4f3Zq5WKVZowx9TCBxmImNJ1qmUi13pB8otwM_l5lfY1AFBMxVbQCUXntLovhDaiSvYp4wGDjFzQiYA-pUq8h6MUZBnhleYrkU7XlCBwNVyN8qNMkpLA7KFZYz-486GnV2NIJJx_4BGa3HdKwQGxi2tjuQsQvao5W4xmSVaaEWopBwMy2QmlhSFQuPUpTaywTqUcUq_6SfAHhZ4IDa_FxEd2c2z8gFGtfst9cY3lRYf-c_ZdboY3mqN9Su3-j3z5r2SHWlhB_LNAjyWlBGsvbGPlTqDziYQwZN4aGsqVKQb9Vw",
+      "e": "AQAB"
+    }
+  ]
+}
+)";
+
+const std::string JwksUri = R"(
+uri: https://pubkey_server/pubkey_path
+cluster: pubkey_cluster
+timeout:
+  seconds: 5
+)";
+
+class JwksFetcherTest : public ::testing::Test {
+public:
+  void SetUp() { MessageUtil::loadFromYaml(JwksUri, uri_); }
+  HttpUri uri_;
+  testing::NiceMock<Server::Configuration::MockFactoryContext> mock_factory_ctx_;
+};
+
+// A mock HTTP upstream with response body.
+class MockUpstream {
```

Comment:
Most likely this mock is useful for other jwt test codes. please also move it to mocks.h

Reply:
Fixed in 8038924